frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardPage.tsx

```diff
@@ -0,0 +1,63 @@
+import type { WithRouterProps } from "react-router";
+
+import { useSyncURLSlug } from "metabase/dashboard/components/DashboardTabs/use-sync-url-slug";
+import {
+  useDashboardUrlParams,
+  useRefreshDashboard,
+} from "metabase/dashboard/hooks";
+
+import { PublicOrEmbeddedDashboard } from "./PublicOrEmbeddedDashboard";
+import { usePublicDashboardEndpoints } from "./WithPublicDashboardEndpoints";
+
+export const PublicOrEmbeddedDashboardPage = (props: WithRouterProps) => {
```

Comment:
I don't know if I'm asking too much, but could you double-check if we need to have this block in `routes.jsx` at all?

https://github.com/metabase/metabase/blob/d527f37a7e87081d243bad16b27d60b82879290a/frontend/src/metabase/routes.jsx#L88-L95

We've been modifying this block, possibly without checking if it's even working. We already have `/Users/kelvin/workspace/metabase/frontend/src/metabase/routes-public.jsx` that you modifies that handles public links.

The block I was mentioning in the main route _would_ only work if we navigate to public link from within Metabase, which doesn't seem to be possible from navigation logic.

https://github.com/metabase/metabase/blob/d527f37a7e87081d243bad16b27d60b82879290a/frontend/src/metabase/lib/dom.js#L238-L244

If we can get rid of that block we could also get rid of `PublicOrEmbeddedDashboardControlled` which is only used in the main `routes.jsx`

Reply:
The second thing is I think we missed porting the title logic from the previous component
https://github.com/metabase/metabase/blob/1fa5a933c68ebf76b3f8af9baee00fe1cf319297/frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx#L220-L222

Although it's broken at the moment (I reported this in Slack), I think we should port that logic so it's easy to follow the refactor which should be 1:1 the current behavior.

Reply:
Thanks, I removed them in fbe0fd88be4917647e6a046261ffd74481ce5503, you're right that after the fix for metabase#38640 we don't need those routes anymore 🚀